sphinxcontrib/openapi/renderers/_httpdomain.py

```diff
@@ -315,3 +336,118 @@ def render_response_content(self, media_type, status_code):
             yield f"   Content-Type: {content_type}"
             yield f""
             yield from indented(example.splitlines())
+
+
+def _generate_example_from_schema(schema):
+    """
+    Generates an example request/response body from the provided schema.
+
+    >>> schema = {
+    ...     "type": "object",
+    ...     "required": ["id", "name"],
+    ...     "properties": {
+    ...         "id": {
+    ...             "type": "integer",
+    ...             "format": "int64"
+    ...         },
+    ...         "name": {
+    ...             "type": "string",
+    ...             "example": "John Smith"
+    ...         },
+    ...         "tag": {
+    ...             "type": "string"
+    ...         }
+    ...     }
+    ... }
+    >>> example = _generate_example_from_schema(schema)
+    >>> assert example == {
+    ...     "id": 1,
+    ...     "name": "John Smith",
+    ...     "tag": "string"
+    ... }
+    """
+    # If an example was provided then we use that
+    if "example" in schema:
+        return schema["example"]
```

Comment:
I think this stuff is covered by the renderer itself, when examples are probed in order:

* `examples`
* `example`
* `schema.example`
* *generate-example-from-schema*

That said, I think we don't need this check here and I'd prefer the function to be simply: get schema as input, generate example as output.

Reply:
I cannot find a clear description of it in the OpenAPI spec but the [Swagger docs](https://swagger.io/docs/specification/adding-examples/) (section called "Object and Property Examples") mention that individual object schema properties can have examples. I would like to keep this as it allows a user to partially override a generated example if what would have been generated is not appropriate. Would you mind if I keep this check but I move it into the loop over object properties?

Reply:
Makes sense. Let's keep in then.